Assets/HoloToolkit/Input/Scripts/GestureManager.cs

```diff
@@ -67,34 +76,63 @@ public partial class GestureManager : Singleton<GestureManager>
         public Vector3 ManipulationOffset { get; private set; }
 
         /// <summary>
-        /// The world space position of the hand being used for the current manipulation gesture.  Not valid
-        /// if a manipulation gesture is not in progress.
+        /// The world space position of manipulation source being used for the current manipulation gesture.
+        /// Valid only if a manipulation gesture is in progress.
         /// </summary>
-        public Vector3 ManipulationHandPosition
+        public Vector3 ManipulationPosition
         {
             get
             {
-                Vector3 handPosition = Vector3.zero;
-                currentHandState.properties.location.TryGetPosition(out handPosition);
-                return handPosition;
+                Vector3 position;
+                if (!currentInteractionSourceState.properties.location.TryGetPosition(out position))
+                {
+                    position = Vector3.zero;
+                }
+                return position;
             }
         }
 
+        /// <summary>
+        /// InteractionSourceDetected tracks the interaction detected state.
+        /// Returns true if the list of tracked interactions is not empty.
+        /// </summary>
+        public bool InteractionSourceDetected
+        {
+            get { return trackedInteractionState.Count > 0; }
+        }
+
+        #endregion
+
+#if UNITY_EDITOR || UNITY_STANDALONE
+        /// <summary>
+        /// Key to press that will select the currently focused object.
+        /// </summary>
+        [SerializeField]
+        private KeyCode keyboardSelectKey = KeyCode.Space;
```

Comment:
I would make keybaordSelectKey public. I use this to test gestures using the keyboard or mouse. I have a singleton, similar to manual Camera Control, that simulates gestures in the unity editor; unless we would want to add that kind of code in the GestureManager as well.